ambry-frontend/src/main/java/com.github.ambry.frontend/RestRequestMetricsGroup.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.frontend;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.rest.RestRequestMetrics;
+import com.github.ambry.utils.Pair;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+
+/**
+ * Class to hold different {@link RestRequestMetrics} for plain text, ssl, with and without encryption
+ */
+class RestRequestMetricsGroup {
+  private static final String SSL_SUFFIX = "Ssl";
+  private static final String ENCRYPTED_SUFFIX = "Encrypted";
+
+  private final ConcurrentMap<Pair<String, String>, ContainerMetrics> instantiatedContainerMetrics =
+      new ConcurrentHashMap<>();
+  private final String requestType;
+  private final boolean encryptedMetricsEnabled;
+  private final boolean containerMetricsEnabled;
+  private final MetricRegistry metricRegistry;
+  private final RestRequestMetrics blobMetrics;
```

Comment:
Could call this blobPlainMetrics or blobPlaintextMetrics to make clear that it's not all blob requests.